Detectors/GlobalTrackingWorkflow/tpcinterpolationworkflow/src/tpc-interpolation-workflow.cxx

```diff
@@ -66,25 +66,41 @@ void customize(std::vector<o2::framework::CompletionPolicy>& policies)
 WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
 {
   WorkflowSpec specs;
-  GID::mask_t allowedSources = GID::getSourcesMask("ITS,TPC,TRD,TOF,ITS-TPC,ITS-TPC-TRD,ITS-TPC-TOF,ITS-TPC-TRD-TOF");
+  GID::mask_t allowedSources = GID::getSourcesMask("ITS-TPC,ITS-TPC-TRD,ITS-TPC-TOF,ITS-TPC-TRD-TOF");
+  GID::mask_t srcVtx = allowedSources & GID::getSourcesMask(configcontext.options().get<std::string>("vtx-sources"));
+  GID::mask_t srcTracks = allowedSources & GID::getSourcesMask(configcontext.options().get<std::string>("tracking-sources"));
+  if (srcTracks.count() > srcVtx.count()) {
+    LOGP(error, "More sources configured for inter-/extrapolation: {} than for vertexing: {}. Additional sources will be ignored", GID::getSourcesNames(srcTracks), GID::getSourcesNames(srcVtx));
+    srcTracks &= srcVtx;
+  }
+  LOG(debug) << "Data sources for inter-/extrapolation: " << GID::getSourcesNames(srcTracks);
+  // check first if ITS-TPC tracks were specifically requested from command line
+  bool processITSTPConly = (srcTracks & GID::getSourcesMask("ITS-TPC")) == GID::getSourcesMask("ITS-TPC");
```

Comment:
for the record: one can do simply `bool processITSTPConly = srcTracks[GID::ITSTPC];`

Reply:
thanks, will change